File: langs/js/utils/from_file.js
const fs = require('fs');
class FromFile{
  set(ss){
    if(ss.length == 0){ this.empty = true; this.ss = '';}
    else { this.empty = false; this.ss = ss;}
  }
  all(){ return this.ss;}
  list(){ let ss = this.ss.split("\r\n"); ss = ss.map(s => s.trim()); let l = ss.pop(); if(l){ss.push(l);} return ss;}
}

function ffUser(path){ const t = fs.readFileSync(path, "utf-8"), ff = new FromFile(); ff.set(t); return ff;}
module.exports = ffUser;
if(require.main === module){
  const issue1 = "a:/pj/sc/lib/rs/issues/issue1.txt", res = ffUser(issue1)
  console.log(res.all());
}

File: langs/ocaml/utils/from_file.ml
#use "a:/PJ/SC/langs/OCaml/utils/print_util.ml"
#use "a:/PJ/SC/langs/OCaml/utils/to_wraper.ml"
#use "a:/PJ/SC/langs/OCaml/utils/to_string.ml"

let read_to_file path = let chan = open_in path in let buf = Buffer.create 4096 in
  let rec loop () = let line = input_line chan in Buffer.add_string buf line; Buffer.add_char buf '\n'; loop () in
    try loop() with End_of_file -> Buffer.contents buf

let ffUser path = let ls = lines(read_to_file path) in List.map(fun l -> String.trim l) ls

let f_f_test go = let issue1 = "a:/pj/sc/lib/rs/issues/issue2.txt"in let ans = ffUser issue1 in ps (sSL ans)

(* let() = f_f_test "go"*)
